packages/react-email/src/components/theme-context.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+
+export type Theme = 'light' | 'dark';
+
+interface ThemeContextType {
+  theme: Theme;
+  setTheme: (theme: Theme) => void;
+}
+
+export const ThemeContext = React.createContext<ThemeContextType | undefined>(
+  undefined,
+);
+
+interface ThemeProviderProps {
+  children: React.ReactNode;
+  theme?: string;
+  setTheme?: (theme: string) => void;
+}
+
+export const ThemeProvider = ({
+  children,
+  theme: externalTheme,
+  setTheme: externalSetTheme,
+}: ThemeProviderProps) => {
+  // Internal state as fallback when props aren't provided
+  const [internalTheme, setInternalTheme] = React.useState<Theme>('light');
+
+  // Use external or internal state
+  const theme = externalTheme ?? internalTheme;
+  const setTheme = externalSetTheme ?? setInternalTheme;
+
+  React.useEffect(() => {
+    const root = document.documentElement;
```

Comment:
We should probably be setting this inside of the document of the preview's iframe instead of in the entire document of the preview server, since that might be problematic

Reply:
Yeah I've added a new hook to do this. :D